CMakeLists.txt

```diff
@@ -58,17 +58,16 @@ file(GLOB AWS_ARCH_SRC
         )
 
 if (USE_CPU_EXTENSIONS)
-    if (MSVC AND AWS_ARCH_INTEL)
+    # First, check if inline assembly is available. Inline assembly can also be supported by MSVC if the compiler in use is Clang.
+    if(AWS_ARCH_INTEL AND AWS_HAVE_GCC_INLINE_ASM)
+        file(GLOB AWS_ARCH_SRC
+                "source/intel/asm/*.c"
+            )
+    elseif (MSVC AND AWS_ARCH_INTEL)
```

Comment:
nitpick: I realize the code was already like this, but I find this logic hard to follow:
```
if INTEL AND ASM:
    ...
else if MSVC AND INTEL:
    ...
```

can we change this to:
```
if INTEL:
    if ASM:
        ...
    else if MSVC:
        ...
```

Reply:
Thanks, updated.